geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/data/KafkaDataStore.scala

```diff
@@ -386,6 +444,21 @@ object KafkaDataStore extends LazyLogging {
     }
   }
 
+  /**
+   * Create a layer view based on a config and the actual feature type
+   *
+   * @param sft simple feature type the view is based on
+   * @param config layer view config
+   * @return
+   */
+  private[kafka] def createLayerView(sft: SimpleFeatureType, config: LayerViewConfig): LayerView = {
+    val viewSft = SimpleFeatureTypes.renameSft(sft, config.typeName)
+    val filter = config.filter.map(FastFilterFactory.optimize(viewSft, _))
+    val transform = config.transform.map(Transforms(viewSft, _))
+    val finalSft = transform.map(Transforms.schema(viewSft, _)).getOrElse(viewSft)
+    LayerView(finalSft, filter, transform)
```

Comment:
Do we have a cheap/quick way to verify that the filter will work against the `finalSft`?

Reply:
it gets applied to the original sft so there's no need to check.